### PR TITLE
added PHP/Heap Sort issue #2147

### LINKED ISSUE #2147: Heap Sort
- [x] C
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [x] Dart
- [x] Go - @Shr03mink
- [x] Java
- [x] JS
- [ ] Kotlin
- [x] Php - @anmolkaur18
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by anmolkaur18**
Fixes #2147 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-added a new file 

#### Languages Used:
-PHP 

#### Files Added:
-Heap_Sort.php 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by anmolkaur18** [comment]
Please review @mrwonko

File: Heap_Sort/Heap_Sort.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      $data[0] = $temp;
+ 
+      $heapSize--;
+      MaxHeapify($data, $heapSize, 0);
+   }
+}
+ 
+ // example of driver function
+$array = array(20,43,65,88,11,33,56,74);
+HeapSort($array,8);
```
Comment (singh-shreya6) [inline Heap_Sort/Heap_Sort.php]: space after comma for readability in input as well.

File: Heap_Sort/Heap_Sort.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    // Recursively heapify the affected sub-tree 
+      MaxHeapify($data, $heapSize, $largest);
+   }
+}
+ 
+ 
+function HeapSort(&$data, $count) {
+   $heapSize = $count;
+ 
+ // Build heap (rearrange array)
```
Comment (singh-shreya6) [inline Heap_Sort/Heap_Sort.php]: Indent comments in line with the code

**Comment by anmolkaur18** [comment]
I have updated and made the changes. Please check

File: Heap_Sort/Heap_Sort.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     if ($left < $heapSize && $data[$left] > $data[$index])  // If left child is larger than root 
+         $largest = $left;
+     else
+         $largest = $index;
+ 
+     if ($right < $heapSize && $data[$right] > $data[$largest]) // If right child is larger than largest so far 
+         $largest = $right;
+  
+    if ($largest != $index)  // If largest is not root 
+   {
```
Comment (singh-shreya6) [inline Heap_Sort/Heap_Sort.php]: The indentation of the if block is one less i.e 3 spaces. This bracket starts one space before if. Please add one more space before bracket also.

File: Heap_Sort/Heap_Sort.php (hunk 1 above)
Comment (singh-shreya6) [reply Heap_Sort/Heap_Sort.php]: Please add the space here as well around comma.

**Comment by anmolkaur18** [comment]
updated the  PR :)

**Comment by anmolkaur18** [comment]
I have changed the title is this correct ?
Also I am confused about template , do i have to make a new pull request or what ?( Sorry I am not aware of these terms I'm a newbie to open source ) 
@singh-shreya6

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 4]
```diff
@@ -0,0 +1,99 @@
+// Required library to use inbuilt queues in Dart
```
Comment (singh-shreya6) [inline Breadth_First_Search/Breadth_First_Search.dart]: Give a brief problem description as multiline comment.

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import 'dart:collection';
+// Required library to use standard output facility of dart
+import 'dart:io';
+
+// Function to find the breadth first search of a graph
+void Breadth_First_Search( var graph, int source, int no_vertices ) {
+    // Queue to order nodes levelwise or breadthwise
+    var queue = Queue();
+    // A list to keep track of visited nodes
```
Comment (singh-shreya6) [inline Breadth_First_Search/Breadth_First_Search.dart]: Blank line before this line.

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    var visited = new List(no_vertices);
+    // Initializing the list
```
Comment (singh-shreya6) [inline Breadth_First_Search/Breadth_First_Search.dart]: Blank line before this line.

**Comment by singh-shreya6** [comment]
Are you working on this @anmolkaur18

**Comment by anmolkaur18** [comment]
Yes I have fixed the template and also mentioned the issue number

**Comment by singh-shreya6** [comment]
There are 8 files in files changed. Please remove others.

**Comment by singh-shreya6** [comment]
Also the PR template is not fixed. No Issue number, it should go like Fixes#issueNumber
Please either make all the suggested changes at one go, or else I will close this PR.

**Comment by anmolkaur18** [comment]
@singh-shreya6 mam pls review. I have made all changes now.

**Comment by singh-shreya6** [comment]
are you working on this @anmolkaur18 ? 
All changes are done. Please squash your commits. Else will have to close the PR.

**Pull request closed without merging**